java/src/jmri/NamedBean.java

```diff
@@ -76,18 +81,18 @@
      * @param listenerRef - A textual reference for the listener, that can be
      *                    presented to the user when a delete is called
      */
-    public void addPropertyChangeListener(java.beans.PropertyChangeListener l, String name, String listenerRef);
+    public void addPropertyChangeListener(@Nonnull java.beans.PropertyChangeListener l, String name, String listenerRef);
```

Comment:
Its probably better to have some interface that extends ```PropertyChangeListener``` have a method like ```public String getDisplayName()``` and have these methods require that interface (so the HashMaps of listernRefs and listeners and PropertyChangeSupport all don't need to be kept in sync).